### PR TITLE
feat(deposit2): token selector

### PR DESCRIPTION
**Pull request by tinaszheng**
This PR introduces the token select list for the new deposit flow:

https://github.com/user-attachments/assets/6d69ee02-59f2-409b-890f-60b2b61437b2

### CODE REVIEW

File: src/views/dialogs/DepositDialog2/AmountInput.tsx [hunk 1]
```diff
@@ -33,10 +41,27 @@ export const AmountInput = ({ value, onChange, token, onTokenClick }: AmountInpu
...
+          {stringGetter({ key: STRING_KEYS.AMOUNT })}
+
+          {tokenBalance.formatted && (
+            <>
+              <span> • </span>
+              <Output
+                tw="inline font-medium text-color-text-0"
+                fractionDigits={4}
+                slotRight=" held" // TODO(deposit2.0): localization here
+                value={BigNumber(tokenBalance.formatted)}
```
Comment (jaredvu) [inline src/views/dialogs/DepositDialog2/AmountInput.tsx:59]: no need to cast as BigNumber if it is a string?

File: src/views/dialogs/DepositDialog2/AmountInput.tsx [hunk 2]
```diff
@@ -33,10 +41,27 @@ export const AmountInput = ({ value, onChange, token, onTokenClick }: AmountInpu
...
-        <div>{stringGetter({ key: STRING_KEYS.AMOUNT })}</div>
+        <div>
+          {stringGetter({ key: STRING_KEYS.AMOUNT })}
+
+          {tokenBalance.formatted && (
+            <>
+              <span> • </span>
+              <Output
+                tw="inline font-medium text-color-text-0"
+                fractionDigits={4}
```
Comment (jaredvu) [inline src/views/dialogs/DepositDialog2/AmountInput.tsx:57]: `4` -> `TOKEN_DECIMALS`

File: src/views/dialogs/DepositDialog2/queries.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          address: sourceAccount.address!,
+          denoms: [USDC_ADDRESSES[CosmosChainId.Neutron]],
+        },
+        [CosmosChainId.Osmosis]: {
+          address: sourceAccount.address!,
+          denoms: [USDC_ADDRESSES[CosmosChainId.Osmosis]],
+        },
+        [CosmosChainId.Noble]: {
+          address: sourceAccount.address!,
+          denoms: [USDC_ADDRESSES[CosmosChainId.Osmosis]],
```
Comment (jaredvu) [inline src/views/dialogs/DepositDialog2/queries.ts:104]: Wrong chain id?

File: src/views/dialogs/DepositDialog2/TokenSelect.tsx [hunk 4]
```diff
@@ -1,7 +1,143 @@
...
+              </div>
+            </button>
+            {i < withBalances.length - 1 && (
+              <hr tw="w-full border border-solid border-color-border" />
+            )}
+          </Fragment>
+        ))}
+      </div>
       <button type="button" onClick={onBack}>
         Go back
```
Comment (jaredvu) [inline src/views/dialogs/DepositDialog2/TokenSelect.tsx:142]: todo: localization :D

Comment (tinaszheng) [reply src/views/dialogs/DepositDialog2/TokenSelect.tsx:142]: Haha I meant to delete this sorry xD

File: src/views/dialogs/DepositDialog2/queries.ts (hunk 3 above)
Comment (tinaszheng) [reply src/views/dialogs/DepositDialog2/queries.ts:104]: omg good catch

**Review by jaredvu — APPROVED** [review]

**Pull request merged**